tests/test_compare_dictconfig_vs_dict.py

```diff
@@ -0,0 +1,211 @@
+"""
```

Comment:
What do you think about [this style](https://github.com/omry/omegaconf/commit/d53960c30faa420015457d3730a881b8a92aca2f#diff-8ce1623c935afb10aeeecdcb7f6306610f5e6039448c224b1e10faeab33c4267R63-R65) for comparing?

Seems a bit cleaner to me:

```python
@pytest.mark.parametrize("cfg", [{}, OmegaConf.create({}, flags={"struct": True})])
def test_struct_dict_get(cfg: Any) -> None:
    assert cfg.get("z") is None
```

Reply:
Are you saying to use `DictConfig.get` instead of `DictConfig.__getitem__`?

Reply:
Ooh, or do you mean creating the `dict` object and the `DictConfig` object next to eachother in the `parametrize` decorator?

Reply:
Hmm... One issue with that is: some of the tests might modify the `parametrize` data, so other tests using the same data could be affected.

I think that using a fixture to instantiate or deepcopy the data helps to make sure that the test cases are isolated from eachother.

Reply:
> Hmm... One issue with that is: some of the tests might modify the parametrize data, so other tests using the same data could be affected.

Yeah, either deepcopy the data in the test or have the fixture provide a new copy every time.